### PR TITLE
Bazel API improvements

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: plugins/package-managers/bazel/src/main/kotlin/CompositeBazelModuleRegistryService.kt [hunk 1]
```diff
@@ -47,7 +47,9 @@ internal class CompositeBazelModuleRegistryService(
          * created based on the passed in [urls]; local registries use the given [projectDir] as workspace.
          */
         fun create(urls: Collection<String>, projectDir: File): CompositeBazelModuleRegistryService {
-            val packageNamesForServer = urls.filter { it.endsWith("source.json") }.mapNotNull { url ->
+            val registryUrls = urls.distinctBy { it.removeSuffix("/") }
```
Comment (nnobelis) [inline plugins/package-managers/bazel/src/main/kotlin/CompositeBazelModuleRegistryService.kt:50]: I don't understand why you need disambiguation here when there is the `filter` underneath that accepts only URLs ending with `source.json`.

Comment (sschuberth) [reply plugins/package-managers/bazel/src/main/kotlin/CompositeBazelModuleRegistryService.kt:50]: Well, couldn't there still be the same URL ending with "source.json" multiple times in the collection? But anyway I agree that the `removeSuffix("/")` "normalization" can be omitted here.

File: clients/bazel-module-registry/src/main/kotlin/LocalBazelModuleRegistryService.kt
Comment (mnonnenmacher) [inline clients/bazel-module-registry/src/main/kotlin/LocalBazelModuleRegistryService.kt:1]: Commit message: "This only" -> "The only"

File: plugins/package-managers/bazel/src/main/kotlin/BazelModel.kt
Comment (mnonnenmacher) [inline plugins/package-managers/bazel/src/main/kotlin/BazelModel.kt:1]: Commit message:
* Maybe "producing lists"
* "accepting collection**s**"

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

**Review by mnonnenmacher — APPROVED** [review]

**Pull request merged**